File: PWGen.cpp
#include <iostream>
#include <string>
#include <ctime>
#include <cstdlib>

std::string generatePassword(int length = 12) {
    const std::string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+";
    std::string password;
    srand(time(0));

    for (int i = 0; i < length; i++) {
        password += chars[rand() % chars.length()];
    }

    return password;
}

int main() {
    std::cout << generatePassword(16) << std::endl;
    return 0;
}

File: PWGen.rs
use rand::Rng;

fn generate_password(length: usize) -> String {
    const CHARS: &[u8] = b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+";
    let mut rng = rand::thread_rng();
    (0..length)
        .map(|_| CHARS[rng.gen_range(0..CHARS.len())] as char)
        .collect()
}

fn main() {
    let password = generate_password(16);
    println!("{}", password);
}
